hadoop-ozone/ozone-recon/src/test/java/org/apache/hadoop/ozone/recon/spi/impl/TestOzoneManagerServiceProviderImpl.java

```diff
@@ -158,4 +165,162 @@ public void testGetOzoneManagerDBSnapshot() throws Exception {
+
+    ArgumentCaptor<ReconTaskStatus> captor =
+        ArgumentCaptor.forClass(ReconTaskStatus.class);
+    verify(reconTaskStatusDaoMock, times(1))
+        .update(captor.capture());
+    assertTrue(captor.getValue().getTaskName().equals("OM_DB_FULL_SNAPSHOT"));
+    verify(reconTaskControllerMock, times(1))
+        .reInitializeTasks(omMetadataManager);
+  }
+
+  @Test
+  public void testSyncDataFromOMDeltaUpdates() throws Exception {
+
+    // Non-Empty OM DB to start with.
+    ReconOMMetadataManager omMetadataManager = getTestMetadataManager(
+        initializeNewOmMetadataManager());
+    ReconTaskStatusDao reconTaskStatusDaoMock =
+        mock(ReconTaskStatusDao.class);
+    doNothing().when(reconTaskStatusDaoMock)
+        .update(any(ReconTaskStatus.class));
+
+    ReconTaskController reconTaskControllerMock = getMockTaskController();
+    when(reconTaskControllerMock.getReconTaskStatusDao())
+        .thenReturn(reconTaskStatusDaoMock);
+    doNothing().when(reconTaskControllerMock)
+        .consumeOMEvents(any(OMUpdateEventBatch.class),
+            any(OMMetadataManager.class));
+
+    OzoneManagerServiceProviderImpl ozoneManagerServiceProvider =
+        new OzoneManagerServiceProviderImpl(configuration, omMetadataManager,
+            reconTaskControllerMock, new ReconUtils());
+    OzoneManagerProtocol ozoneManagerProtocolMock =
+        mock(OzoneManagerProtocol.class);
+    when(ozoneManagerProtocolMock.getDBUpdates(any(OzoneManagerProtocolProtos
+        .DBUpdatesRequest.class))).thenReturn(new DBUpdatesWrapper());
+    injectOzoneServiceProviderField(ozoneManagerServiceProvider,
+        ozoneManagerProtocolMock, "ozoneManagerClient");
+
+    // Should trigger delta updates.
+    ozoneManagerServiceProvider.syncDataFromOM();
+
+    ArgumentCaptor<ReconTaskStatus> captor =
+        ArgumentCaptor.forClass(ReconTaskStatus.class);
+    verify(reconTaskStatusDaoMock, times(1))
+        .update(captor.capture());
+    assertTrue(captor.getValue().getTaskName().equals("OM_DB_DELTA_UPDATES"));
+
+    verify(reconTaskControllerMock, times(1))
+        .consumeOMEvents(any(OMUpdateEventBatch.class),
+            any(OMMetadataManager.class));
+  }
+
+  private ReconTaskController getMockTaskController() {
+    ReconTaskController reconTaskControllerMock =
+        mock(ReconTaskController.class);
+    return reconTaskControllerMock;
+  }
+
+  private ReconUtils getMockReconUtils() throws IOException {
+    ReconUtils reconUtilsMock = mock(ReconUtils.class);
+    when(reconUtilsMock.getReconDbDir(any(), anyString())).thenCallRealMethod();
+    doCallRealMethod().when(reconUtilsMock).untarCheckpointFile(any(), any());
+    return reconUtilsMock;
+  }
+
+  private void injectOzoneServiceProviderField(
+      OzoneManagerServiceProviderImpl ozoneManagerServiceProvider,
+      Object fieldValue, String fieldName)
+      throws NoSuchFieldException, IllegalAccessException {
+    Field f1 = ozoneManagerServiceProvider.getClass()
```

Comment:
Let's use the @Provides method for some stateful instance creation instead of ugly refection code :-)